Add Py36 environments to travis

francisco-dlp opened the pull request:
This also uses miniconda for the minimum environment instead of the system python as it was before to make the code more compact.

This is WIP and is currently broken. travis is having issues today and the log is currently unavailable, so I don't know why it's broken. Any help would be appreciated.

francisco-dlp committed:
Automatic style corrections courtesy of autopep8

francisco-dlp committed:
Update CHANGES and some docs

francisco-dlp committed:
v1.2

francisco-dlp committed:
Merge remote-tracking branch 'upstream/RELEASE_next_minor' into RELEASE_v1.2

francisco-dlp committed:
1.1.3 -> 1.2

francisco-dlp committed:
Some fixes to docs

francisco-dlp committed:
Merge remote-tracking branch 'upstream/RELEASE_next_minor' into RELEASE_v1.2

francisco-dlp committed:
Automatic style corrections courtesy of autopep8

francisco-dlp committed:
Merge branch 'RELEASE_next_minor' into RELEASE_v1.2

francisco-dlp committed:
Fix formatting

francisco-dlp committed:
Add min scipy version

francisco-dlp committed:
Install with pip in appveyour winpython deploy

francisco-dlp committed:
Test hyperspyui inclusion

francisco-dlp committed:
Test hyperspyui inclusion

francisco-dlp committed:
Revert "Test hyperspyui inclusion"

This reverts commit dfc3a550c883aa78013506348bc47d156b421c2e.

francisco-dlp committed:
Try upgrading packages

francisco-dlp committed:
Temporarily disable test and 32bit for quicker build

francisco-dlp committed:
Upgrade WP pip

francisco-dlp committed:
Revert "Upgrade WP pip"

This reverts commit c98ff899815473504511a39b705a3a807db22e31.

francisco-dlp committed:
Upgrade only selected packages

francisco-dlp committed:
Upgrade only selected packages 2

francisco-dlp committed:
Configuration works, uncomment lines

francisco-dlp committed:
Set dask version to 0.13 as 0.14 is possibly buggy (breaks tests)

francisco-dlp committed:
fix_indentation

francisco-dlp committed:
Uncomment running tests

francisco-dlp committed:
Merge remote-tracking branch 'upstream/RELEASE_next_minor' into RELEASE_v1.2

francisco-dlp committed:
Avoid dask 0.14

francisco-dlp committed:
Add Py36 to appveyor

francisco-dlp committed:
Attempt to fix memory issue

francisco-dlp committed:
Merge branch 'RELEASE_next_minor' of https://github.com/hyperspy/hyperspy into RELEASE_v1.2

francisco-dlp committed:
Automatically change to agg backend when no xserver available

francisco-dlp committed:
No need to skip plot tests in macOS

francisco-dlp committed:
Merge branch 'RELEASE_v1.2' of https://github.com/francisco-dlp/hyperspy into RELEASE_v1.2

francisco-dlp committed:
Set backend to agg in macOS

francisco-dlp committed:
Let 5 pixels fail randomly

francisco-dlp committed:
Merge branch 'RELEASE_v1.2' of https://github.com/francisco-dlp/hyperspy into RELEASE_v1.2

francisco-dlp committed:
Fix test

francisco-dlp committed:
Merge remote-tracking branch 'upstream/RELEASE_next_minor' into RELEASE_v1.2

francisco-dlp committed:
Fix deprecated extension

francisco-dlp committed:
Add hspy extension and some fixes to the docs

francisco-dlp committed:
Merge remote-tracking branch 'upstream/RELEASE_next_minor' into RELEASE_v1.2

francisco-dlp committed:
Added valuemin and indexmin to changes and user guide

francisco-dlp committed:
Add lazy valuemin

francisco-dlp committed:
Add Py36 build to travis and min env with miniconda

francisco-dlp committed:
Fix typo

francisco-dlp committed:
Remove all but linux py36 env for testing

francisco-dlp committed:
Build ext

francisco-dlp committed:
Revert "Remove all but linux py36 env for testing"

This reverts commit 07031ab20d6bb6f24031dc1e45adc73f1437def4.

francisco-dlp committed:
Build the sdist ones

francisco-dlp committed:
remove unnecessary (?) export

francisco-dlp committed:
Revert "remove unnecessary (?) export"

This reverts commit a15fbb9e4cc9abf8b91a4ff289216da6821fd072.

francisco-dlp committed:
Fix env header?

francisco-dlp committed:
Test in the dark (travis log broken)

francisco-dlp committed:
Test in the dark (travis log broken)

francisco-dlp committed:
Revert "Test in the dark (travis log broken)"

This reverts commit 9c8c3a20fcc4ef9397430fe92889e8fe36dbc7a3.

francisco-dlp committed:
Test in the dark (travis log broken)

francisco-dlp committed:
Revert commit range

francisco-dlp committed:
Test in the dark (travis log broken)

francisco-dlp committed:
Revert "Test in the dark (travis log broken)"

This reverts commit f3f326b2df28015f3c13b0eb7c4d1be7371cf57c.

francisco-dlp committed:
Test in the dark (travis log broken)

francisco-dlp committed:
Quick test

francisco-dlp committed:
Debug min env

francisco-dlp committed:
Revert "Debug min env"

This reverts commit ac807863e9d904594741f4808c461dcab9c1cf61.

francisco-dlp committed:
Fix allow failures

sem-geologist commented:
Do we really need such huge matrix?
min env should not go with miniconda, it kills completely the purpoise of minimal python. Are we moving completely to py36? Did conda moved there? debian testing is still on 35.

sem-geologist commented on .travis.yml:
```diff
@@ -2,82 +2,61 @@ language: python
-    else pip install pytest;
-    pip3 install .;
+      DEPS="pip numpy scipy matplotlib ipython h5py sympy scikit-learn dill natsort setuptools scikit-image cython lxml ipyparallel dask traits traitsui";
+    else DEPS="pip ipython numpy scipy matplotlib h5py sympy scikit-image";
```
this make no sense, why we need to install `ipython numpy scipy matplotlib h5py` in the minimal environment?
We should not do that – instead of it let to pull the things which are declared in setup.py by installation script.
Else we are only pretending that we test it on minimal environment. We want to emulate exact environment as the customer with his/her python3 version (be it conda or provided with system), if we decide not to test minimal environment then we can get rid of half of this logic.
And if it is tests which fails, then it is tests which have to be fixed.

sem-geologist commented on .travis.yml:
```diff
@@ -2,82 +2,61 @@ language: python
-    else python setup.py sdist;
-    fi;
+    python setup.py bdist_wheel;
+  else python setup.py sdist;
```
No!, why we would build sdist from min environment??? it would have no cython, and we will get no C code prepared for installation from source. This line should go away completely, or be inserted somethere else

sem-geologist commented on .travis.yml:
```diff
@@ -2,82 +2,61 @@ language: python
-    export DISTRIB=$(ls ./dist/*.whl);
-  else export DISTRIB=$(ls ./dist/*.tar.gz);
-  fi
+- export DISTRIB=$(ls ./dist/*.whl);
```
no more sdist?

sem-geologist commented on .travis.yml:
```diff
@@ -2,82 +2,61 @@ language: python
+    language: generic
+  - env: export PYTHON=3.6; FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='False'
+  - env: export PYTHON=3.6; FAIL_ON_EXTERNAL_DEPRECATION='True'; MINIMAL_ENV='False'
+  - env: export PYTHON=3.6; FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='True'
```
if we use miniconda it kill completely the sense of minimal environment. miniconda is however ('spam', 'spam', 'spam'...) not only 'eggs'

sem-geologist commented:
by using miniconda for minimal environment we shave few lines however:
1. prolong the minimal env testing 100%; instead with stock python we could test in the same time both min py36 and min py35 (which is still used on most of the stable distros or LTR releases)
2. by not using stock python we no more emulate pip hyperspy installation on brand new system

francisco-dlp commented:
I think that installing in miniconda is "stricter" than installing in Ubuntu's system Python. Ubuntu comes with more python packages pre-installed than miniconda so, if it installs in miniconda it has to install in Ubuntu. The only issue is that the system pip version may be too old in some systems, but that's not something up to us to fix.

We can test also Py35 using miniconda if we want to (we do it for the standard test), I just thought that it may not be necessary for the min env.

By the way, the min env was an excellent idea!

sem-geologist commented:
but in this shape as it is it is broken, it is pretending to be minimal.
At least remove `DEPS="pip ipython numpy scipy matplotlib h5py sympy scikit-image";`
 this should not be installed through pip, because then we are no more testing what is minimal environment. Those which is minimal should be declared in setup.

sem-geologist commented:
also do not create sdist with minimal environment especialy that the artifacts are not collected in current state of the code anymore

francisco-dlp commented:
I fully agree, maybe we could add an extra command to setup.py to return the requirements so that we can pass them to conda? (We don't want to compile numpy, scipy etc ourselves...)

I modified this in a rush for the HyperSpy 1.2 release and this has already been sort-of merged due to a mistake caused by working from local and remote terminals at the same time: I added the v1.2 tag to a commit in this branch and pushed to upstream thinking that I was in RELEASE_v1.2 :( Anyway, the tests work and we can carry on improving from here.

francisco-dlp commented:
I'll merge it as it has already been effectively merged.

sem-geologist commented:
I probably missed something, but you don't need to compile anything anymore, binary wheels are used in linux already too. So pip does not need to recompile stuff.

francisco-dlp commented:
We have linux wheels in the GH release, but I couldn't upload them to pypi, I got a platform not supported error...

sem-geologist commented:
Not talking about hyperspy wheels, I am talking abou numpy, scipy, ....
wheels, and that it needs no compiling, and so conda is not needed. Even if
it was compiling, previously minimal env was doing everything in half of
time than now with conda and not compiling

On 2 Mar 2017 19:56, "Francisco de la Peña" <notifications@github.com>
wrote:

> We have linux wheels in the GH release, but I couldn't upload them to
> pypi, I got a platform not supported error...
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment-283744893>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALwM5PIXvG-wOVZUmXSndZhwwH70_FZDks5rhxDCgaJpZM4MPkLK>
> .
>

francisco-dlp commented:
You are right, pypi has wheels for all those requirements for linux. Would
you like to send a PR turning back to installing all from pip? By the way,
I don't know why pypi complains when I try to upload hyperspy's linux
wheels.

On 2 March 2017 at 20:20, Petras Jokubauskas <notifications@github.com>
wrote:

> Not talking about hyperspy wheels, I am talking abou numpy, scipy, ....
> wheels, and that it needs no compiling, and so conda is not needed. Even if
> it was compiling, previously minimal env was doing everything in half of
> time than now with conda and not compiling
>
> On 2 Mar 2017 19:56, "Francisco de la Peña" <notifications@github.com>
> wrote:
>
> > We have linux wheels in the GH release, but I couldn't upload them to
> > pypi, I got a platform not supported error...
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment-283744893>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/ALwM5PIXvG-
> wOVZUmXSndZhwwH70_FZDks5rhxDCgaJpZM4MPkLK>
>
> > .
> >
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment-283768389>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA8aF6Gmz6lLlRLmZ2rBH9ImqM7vaXUjks5rhySKgaJpZM4MPkLK>
> .
>

sem-geologist commented:
I see one very obvious thing - scipy, numpy are bdist `manylinux1`, and
hyperspy is `linux`, thats why it is incompatible.

On 2 Mar 2017 21:33, "Francisco de la Peña" <notifications@github.com>
wrote:

> You are right, pypi has wheels for all those requirements for linux. Would
> you like to send a PR turning back to installing all from pip? By the way,
> I don't know why pypi complains when I try to upload hyperspy's linux
> wheels.
>
> On 2 March 2017 at 20:20, Petras Jokubauskas <notifications@github.com>
> wrote:
>
> > Not talking about hyperspy wheels, I am talking abou numpy, scipy, ....
> > wheels, and that it needs no compiling, and so conda is not needed. Even
> if
> > it was compiling, previously minimal env was doing everything in half of
> > time than now with conda and not compiling
> >
> > On 2 Mar 2017 19:56, "Francisco de la Peña" <notifications@github.com>
> > wrote:
> >
> > > We have linux wheels in the GH release, but I couldn't upload them to
> > > pypi, I got a platform not supported error...
> > >
> > > —
> > > You are receiving this because you commented.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment-283744893
> >,
> > > or mute the thread
> > > <https://github.com/notifications/unsubscribe-auth/ALwM5PIXvG-
> > wOVZUmXSndZhwwH70_FZDks5rhxDCgaJpZM4MPkLK>
> >
> > > .
> > >
> >
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment-283768389>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/
> AA8aF6Gmz6lLlRLmZ2rBH9ImqM7vaXUjks5rhySKgaJpZM4MPkLK>
> > .
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment-283771795>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALwM5Ew91IPF-gO7hPI693LIdHxCU9CUks5rhyeygaJpZM4MPkLK>
> .
>

sem-geologist commented:
It actually is the case, I am trying to find how to bdist  to `manylinux1`,
is there any code which push hyperspy bdist artefacts to pypi?

On 2 Mar 2017 22:45, "Petras Jokubauskas" <klavishas@gmail.com> wrote:

> I see one very obvious thing - scipy, numpy are bdist `manylinux1`, and
> hyperspy is `linux`, thats why it is incompatible.
>
> On 2 Mar 2017 21:33, "Francisco de la Peña" <notifications@github.com>
> wrote:
>
>> You are right, pypi has wheels for all those requirements for linux. Would
>> you like to send a PR turning back to installing all from pip? By the way,
>> I don't know why pypi complains when I try to upload hyperspy's linux
>> wheels.
>>
>> On 2 March 2017 at 20:20, Petras Jokubauskas <notifications@github.com>
>> wrote:
>>
>> > Not talking about hyperspy wheels, I am talking abou numpy, scipy, ....
>> > wheels, and that it needs no compiling, and so conda is not needed.
>> Even if
>> > it was compiling, previously minimal env was doing everything in half of
>> > time than now with conda and not compiling
>> >
>> > On 2 Mar 2017 19:56, "Francisco de la Peña" <notifications@github.com>
>> > wrote:
>> >
>> > > We have linux wheels in the GH release, but I couldn't upload them to
>> > > pypi, I got a platform not supported error...
>> > >
>> > > —
>> > > You are receiving this because you commented.
>> > > Reply to this email directly, view it on GitHub
>> > > <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment
>> -283744893>,
>> > > or mute the thread
>> > > <https://github.com/notifications/unsubscribe-auth/ALwM5PIXvG-
>> > wOVZUmXSndZhwwH70_FZDks5rhxDCgaJpZM4MPkLK>
>> >
>> > > .
>> > >
>> >
>> > —
>> > You are receiving this because you modified the open/close state.
>> > Reply to this email directly, view it on GitHub
>> > <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment-283768389
>> >,
>> > or mute the thread
>> > <https://github.com/notifications/unsubscribe-auth/AA8aF6Gmz
>> 6lLlRLmZ2rBH9ImqM7vaXUjks5rhySKgaJpZM4MPkLK>
>> > .
>> >
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment-283771795>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ALwM5Ew91IPF-gO7hPI693LIdHxCU9CUks5rhyeygaJpZM4MPkLK>
>> .
>>
>

francisco-dlp commented:
We push them using twine. It is not done automatically as it relies on a
password.

On 2 March 2017 at 22:18, Petras Jokubauskas <notifications@github.com>
wrote:

> It actually is the case, I am trying to find how to bdist to `manylinux1`,
> is there any code which push hyperspy bdist artefacts to pypi?
>
> On 2 Mar 2017 22:45, "Petras Jokubauskas" <klavishas@gmail.com> wrote:
>
> > I see one very obvious thing - scipy, numpy are bdist `manylinux1`, and
> > hyperspy is `linux`, thats why it is incompatible.
> >
> > On 2 Mar 2017 21:33, "Francisco de la Peña" <notifications@github.com>
> > wrote:
> >
> >> You are right, pypi has wheels for all those requirements for linux.
> Would
> >> you like to send a PR turning back to installing all from pip? By the
> way,
> >> I don't know why pypi complains when I try to upload hyperspy's linux
> >> wheels.
> >>
> >> On 2 March 2017 at 20:20, Petras Jokubauskas <notifications@github.com>
> >> wrote:
> >>
> >> > Not talking about hyperspy wheels, I am talking abou numpy, scipy,
> ....
> >> > wheels, and that it needs no compiling, and so conda is not needed.
> >> Even if
> >> > it was compiling, previously minimal env was doing everything in half
> of
> >> > time than now with conda and not compiling
> >> >
> >> > On 2 Mar 2017 19:56, "Francisco de la Peña" <notifications@github.com
> >
> >> > wrote:
> >> >
> >> > > We have linux wheels in the GH release, but I couldn't upload them
> to
> >> > > pypi, I got a platform not supported error...
> >> > >
> >> > > —
> >> > > You are receiving this because you commented.
> >> > > Reply to this email directly, view it on GitHub
> >> > > <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment
> >> -283744893>,
>
> >> > > or mute the thread
> >> > > <https://github.com/notifications/unsubscribe-auth/ALwM5PIXvG-
> >> > wOVZUmXSndZhwwH70_FZDks5rhxDCgaJpZM4MPkLK>
> >> >
> >> > > .
> >> > >
> >> >
> >> > —
> >> > You are receiving this because you modified the open/close state.
> >> > Reply to this email directly, view it on GitHub
> >> > <https://github.com/hyperspy/hyperspy/pull/1501#
> issuecomment-283768389
> >> >,
> >> > or mute the thread
> >> > <https://github.com/notifications/unsubscribe-auth/AA8aF6Gmz
> >> 6lLlRLmZ2rBH9ImqM7vaXUjks5rhySKgaJpZM4MPkLK>
> >> > .
> >> >
> >>
> >> —
> >> You are receiving this because you commented.
> >> Reply to this email directly, view it on GitHub
> >> <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment-283771795
> >,
> >> or mute the thread
> >> <https://github.com/notifications/unsubscribe-auth/ALwM5Ew91IPF-
> gO7hPI693LIdHxCU9CUks5rhyeygaJpZM4MPkLK>
> >> .
> >>
> >
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyperspy/hyperspy/pull/1501#issuecomment-283800812>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA8aF7kFy1lYg_v5zO5eyjeR4pOWD8NTks5rh0BBgaJpZM4MPkLK>
> .
>

sem-geologist commented:
btw there is the  pep 513 dedicated for the problem [https://www.python.org/dev/peps/pep-0513/#the-manylinux1-policy](https://www.python.org/dev/peps/pep-0513/#the-manylinux1-policy)

sem-geologist commented:
there is auditwheel tool which can be used to create manylinux1 from linux1 (if it is compliant) https://github.com/pypa/auditwheel
